src/python_testing/TC_SU_2_7.py

```diff
@@ -0,0 +1,466 @@
+                     "Verify that the OTA-Subscriber receives a StateTransition event notification for the state change to DelayedOnUserConsent."),
+            TestStep(5, "Force an error during the download of the OTA image to the DUT. Wait for the Idle timeout which should be no less than 5 minutes.", "Verify that the OTA-Subscriber receives a StateTransition event notification for the state change to Idle."
+                     "Verify that the OTA-Subscriber receives a DownloadError event notification on BDX Idle timeout."
+                     "Verify that the data in this event has the following."
+                     "SoftwareVersion - Set to the value of the SoftwareVersion being downloaded."
+                     "BytesDownloaded - Number of bytes that have been downloaded."
+                     "ProgressPercent - Nearest Integer percent value reflecting how far within the transfer the failure occurred. IF the total length of the transfer is unknown, the value can be NULL."
+                     "PlatformCode - Internal product-specific error code or NULL."),
+            TestStep(6, "After the OTA image is transferred, DUT sends ApplyUpdateRequest to the OTA-P. OTA-P/TH sends the ApplyUpdateResponse Command to the DUT. Action field is set to \"AwaitNextAction\".",
+                     "Verify that the OTA-Subscriber receives a StateTransition event notification for the state change to DelayedOnApply."),
+        ]
+
+    @async_test_body
+    async def test_TC_SU_2_7(self):
+        # Requestor is the DUT
+        # Requestor has the flag --autoApply
+        self.step(0)
+        controller = self.default_controller
+        # # Set the time interval based on the ota_image_download_timeout
+        max_interval = self.ota_image_download_timeout + 60
+
+        self.step(1)
+        # Create event subscriber for StateTransition
+        update_state_attr_handler = AttributeSubscriptionHandler(
+            expected_cluster=Clusters.OtaSoftwareUpdateRequestor,
+            expected_attribute=Clusters.OtaSoftwareUpdateRequestor.Attributes.UpdateState
+        )
+        await update_state_attr_handler.start(dev_ctrl=controller, node_id=self.requestor_node_id, endpoint=0,
+                                              fabric_filtered=False, min_interval_sec=0, max_interval_sec=5)
+        state_transition_event_handler = EventSubscriptionHandler(
+            expected_cluster=self.ota_req, expected_event_id=self.ota_req.Events.StateTransition.event_id)
+        await state_transition_event_handler.start(controller, self.requestor_node_id, endpoint=0, min_interval_sec=0, max_interval_sec=max_interval)
+        await self.announce_ota_provider(controller, self.provider_node_id, self.requestor_node_id)
+        # Register event, should change to Querying
+        event_report = state_transition_event_handler.wait_for_event_report(self.ota_req.Events.StateTransition, timeout_sec=60)
```

Comment:
DUT has 600s to query per the spec. So the timeout here needs to be 600s + some fudge